gene/pointMutation.go

```diff
@@ -0,0 +1,129 @@
+package gene
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/numbers"
+)
+
+// PointMutation changes a single nucleotide to the desired base, predicts the effect,
+// and updates the Gene struct to reflect the change.
+// The position of the mutation should be given in base-zero genomic coordinates.
+func PointMutation(g *Gene, genomePos int, alt dna.Base) (EffectPrediction, error) {
+	var answer EffectPrediction
+	answer.StopDist = -1
+	var err error
+
+	err = pointPreRunChecks(g, genomePos, &alt)
+	if err != nil {
+		return EffectPrediction{}, err
+	}
+
+	genomeIndexPos := numbers.AbsInt(genomePos - g.startPos) // abs needed to handle negative posStrand
+	g.genomeSeq[genomeIndexPos] = alt
+
+	cdnaIndexPos := int(g.featureArray[genomeIndexPos])
+
+	if cdnaIndexPos >= 0 { // mutation is coding
+		err = pointCoding(g, cdnaIndexPos, alt, &answer)
+	} else { // mutation is noncoding
+		err = pointNonCoding(g, genomePos, &answer)
+	}
+
+	if err != nil {
+		return EffectPrediction{}, err
+	}
+
+	g.protSeq = dna.TranslateSeq(g.codingSeq.seq) // TODO improve efficiency by updating the protein as changes are made
+	return answer, nil
+}
+
+// checkSplice inputs a cDNA offset value and determines if a variant at this site may affect splicing.
+func checkSplice(CdnaOffset int) MutationType {
+	if numbers.AbsInt(CdnaOffset) <= 2 {
+		return Splice
+	}
+
+	if numbers.AbsInt(CdnaOffset) <= 10 {
+		return FarSplice
+	}
+
+	return Intronic
+}
+
+// pointPreRunChecks performs error checking and appends the diff log prior to making a point mutation.
+func pointPreRunChecks(g *Gene, genomePos int, alt *dna.Base) error {
+	var log diff
+
+	genomeIndexPos := numbers.AbsInt(genomePos - g.startPos) // abs needed to handle negative posStrand
+
+	// Fill log before change
+	log.genomePos = genomePos
+	log.removed = make([]dna.Base, 1)
+	log.removed[0] = g.genomeSeq[genomeIndexPos]
+	log.added = make([]dna.Base, 1)
+	log.added[0] = *alt
+
+	if !g.posStrand { // so that diff can be undone by another PointMutation call
+		dna.ReverseComplement(log.removed)
+	}
+
+	if !dna.IsSeqOfACTG([]dna.Base{*alt}) {
+		return ErrNonACGTBase
+	}
+
+	if genomePos < 0 {
+		return ErrNegativeInputValue
+	}
+	if g.posStrand {
```

Comment:
I am seeing the checks here that I was expecting at the conversion step.  This addresses my previous comments.  I do wonder if making this coordinate conversion its own function (since I have already seen it called a couple times) that included the checks would be more readable.